libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -220,7 +220,12 @@ void AC_PrecLand::update(float rangefinder_alt_cm, bool rangefinder_alt_valid)
 
 bool AC_PrecLand::target_acquired()
 {
-    _target_acquired = _target_acquired && (AP_HAL::millis()-_last_update_ms) < 2000;
+    if ((AP_HAL::millis()-_last_update_ms) > LANDING_TARGET_TIMEOUT_MS) {
```

Comment:
this seems fine but given total freedom I would make target_acquired() a const function and set the _target_acquired in the "run_estimator" method instead.